tasks/EchoesToFrameConverterTask.cpp

```diff
@@ -0,0 +1,172 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "EchoesToFrameConverterTask.hpp"
+#include <base-logging/Logging.hpp>
+#include <base/samples/RigidBodyState.hpp>
+#include <math.h>
+#include <opencv2/imgproc/imgproc.hpp>
+
+using namespace radar_base;
+using namespace cv;
+using namespace base::samples::frame;
+EchoesToFrameConverterTask::EchoesToFrameConverterTask(std::string const& name)
+    : EchoesToFrameConverterTaskBase(name)
+{
+}
+
+EchoesToFrameConverterTask::~EchoesToFrameConverterTask()
+{
+}
+
+/// The following lines are template definitions for the various state machine
+// hooks defined by Orocos::RTT. See EchoesToFrameConverterTask.hpp for more detailed
+// documentation about them.
+
+bool EchoesToFrameConverterTask::configureHook()
+{
+    if (!EchoesToFrameConverterTaskBase::configureHook())
+        return false;
+    configureOutput();
+    updateLookUpTable();
+    return true;
+}
+bool EchoesToFrameConverterTask::startHook()
+{
+    if (!EchoesToFrameConverterTaskBase::startHook())
+        return false;
+    m_last_sample = base::Time::now();
+    return true;
+}
+void EchoesToFrameConverterTask::updateHook()
+{
+    EchoesToFrameConverterTaskBase::updateHook();
+
+    double yaw_correction = 0;
+    base::samples::RigidBodyState sensor2ref_pose;
+    if (_sensor2ref_pose.read(sensor2ref_pose) != RTT::NoData) {
+        yaw_correction = sensor2ref_pose.getYaw();
```

Comment:
If he doesn't have the Yaw information its considered 0? Shouldn't the hook finish?